lua/telekasten.lua

```diff
@@ -201,34 +201,44 @@ local function check_dir_and_ask(dir, purpose)
                     vim.cmd('echomsg " "')
                     vim.cmd('echomsg "' .. dir .. ' created"')
                     ret = true
+                    callback(ret)
                 else
                     -- unreachable: plenary.Path:mkdir() will error out
                     tkutils.print_error("Could not create directory " .. dir)
                     ret = false
+                    callback(ret)
                 end
             end
         end)
     else
         ret = true
+        if callback ~= nil then
+            callback(ret)
```

Comment:
even though `vim.ui.select` was not called in this branch, code still needs to continue through callback